roles/ceph-common/tasks/checks/check_mandatory_vars.yml

```diff
@@ -71,3 +71,20 @@
     - monitor_interface == 'interface'
     - monitor_address == "0.0.0.0"
     - mon_group_name in group_names
+
+- name: "make sure monitor_address_block is valid"
+  set_fact:
+    monitor_address_block_check: "{{ hostvars[item]['ansible_all_ipv4_addresses'] | ipaddr(monitor_address_block) }}"
+  when:
+    - not containerized_deployment
+    - not containerized_deployment_with_kv
+    - monitor_address_block is defined
+  with_items:
+    - "{{ groups[mon_group_name] }}"
+
+- name: "fail if monitor_address_block is invalid"
+  fail:
+    msg: "An IP address in the block {{ monitor_address_block }} was not found on one or more monitor nodes. Please check your monitor_address_block setting."
+  when:
+    - monitor_address_block_check is defined
+    - monitor_address_block_check | length == 0
```

Comment:
First of all, this won't work if `ipaddr` is not installed, see: http://docs.ansible.com/ansible/latest/playbooks_filters_ipaddr.html

Assuming it is, can you try:

```
when:
  - monitor_address_block is defined
  - not item | ipaddr
 with_items: "{{ monitor_address_block }}"
```

This will fail if one of the IP has an invalid format.

Reply:
@fullerdj you didn't adress @leseb's comment, is that normal?